File: Linguagem de Programação I/Extras/Beecrowd_Extra/1124.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            int L = scanner.nextInt();
            int C = scanner.nextInt();
            int R1 = scanner.nextInt();
            int R2 = scanner.nextInt();
            if (L == 0 && C == 0 && R1 == 0 && R2 == 0)
                break;

            if (2 * Math.max(R1, R2) > Math.min(L, C)) {
                System.out.println("N");
                continue;
            }

            if (Math.pow(R1 + R2, 2) <= Math.pow(L - R1 - R2, 2) + Math.pow(C - R1 - R2, 2))
                System.out.println("S");
            else
                System.out.println("N");
        }
        scanner.close();
    }
}

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1124.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        while (true)
        {
            string input = Console.ReadLine();
            if (string.IsNullOrEmpty(input))
            {
                break;
            }

            string[] inputs = input.Split(' ');
            int L = int.Parse(inputs[0]);
            int C = int.Parse(inputs[1]);
            int R1 = int.Parse(inputs[2]);
            int R2 = int.Parse(inputs[3]);

            if (L == 0 && C == 0 && R1 == 0 && R2 == 0)
            {
                break;
            }

            if (Math.Max(R1, R2) * 2 > Math.Min(L, C))
            {
                Console.WriteLine('N');
                continue;
            }

            if (Math.Pow(R1 + R2, 2) <= Math.Pow(L - R1 - R2, 2) + Math.Pow(C - R1 - R2, 2))
            {
                Console.WriteLine('S');
            }
            else
            {
                Console.WriteLine('N');
            }
        }
    }
}